ls1intum/Artemis
# Allow sorting exercises by start date

### Checklist

### Motivation and Context
- The user is only able to sort for dueDates, but it should also be possible to sort for releaseDates

### Description
- Added nur dropdown with a new sortingtype, releaseDate
- added tests for this component
- **Removed code which should be unused, please also double check this in the code review**

### Steps for Testing

1. Log in to Artemis
2. Navigate to Course overview
3. See if exercises are ordered correctly after dueDate
4. Vary with the sorting options (Newest, Oldest, DueDate, ReleaseDate) and make sure exercises are correctly ordered based on the chosen sorting type (see screenshots). Play around with the sorting button and make sure no errors occur and the sorting is always correct.


### Test Coverage
course-exercises.component.ts: 20% -> 97,93%

### Screenshots
<img width="1406" alt="Screenshot 2021-01-15 at 14 02 12" src="https://user-images.githubusercontent.com/44401560/104730487-6e965e00-573a-11eb-8657-6390bc90dbee.png">
